src/main.cpp

```diff
@@ -784,6 +838,42 @@ struct router_thread
 				}
 			}
 
+			if (cross_pollinate
+				&& ep == bootstrap_node)
+			{
+				// this is a response from the other bootstrap node
+				// put the nodes in the response in the ping queue
+				{
+					std::string nodes4 = a.dict_find_string_value("nodes", "");
+					fprintf(stderr, "received %d nodes4 from x-pollinate node\n"
+						, int(nodes4.size() / 26));
+					char const* end = nodes4.data() + nodes4.size();
+					for (char const* i = nodes4.data(); i < end; i += 6 + 20)
```

Comment:
Exit condition should be `end - i >= 6 + 20` so that incorrectly sized responses don't result in reading past the end of the buffer.